app/src/main/java/com/infomaniak/mail/utils/MessageUtils.kt

```diff
@@ -50,5 +50,14 @@ object MessageUtils {
     }
 }
 
-data class ThreadMessageToExecuteAction(val thread: Thread, val messageUid: String)
+sealed class ThreadMessageInteraction(open val messageUid: String) {
+    data class Action(val thread: Thread, override val messageUid: String) : ThreadMessageInteraction(messageUid)
+    data class Reaction(override val messageUid: String): ThreadMessageInteraction(messageUid)
+}
```

Comment:
If your class is only used inside of ThreadActionsViewModel, you can define it there instead

Reply:
Same comment for ThreadMessageToExecuteInteraction actually